spring-cloud-sleuth-instrumentation/src/main/java/org/springframework/cloud/sleuth/instrument/async/LazyTraceScheduledThreadPoolExecutor.java

```diff
@@ -88,32 +88,55 @@
+		this.decorateTaskRunnable = makeAccessibleIfNotNullAndOverridden(decorateTaskRunnable);
+		Method decorateTaskCallable = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "decorateTask",
 				Callable.class, RunnableScheduledFuture.class);
-		makeAccessibleIfNotNull(this.decorateTaskCallable);
-		this.finalize = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "finalize", null);
-		makeAccessibleIfNotNull(this.finalize);
-		this.beforeExecute = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "beforeExecute", null);
-		makeAccessibleIfNotNull(this.beforeExecute);
-		this.afterExecute = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "afterExecute", null);
-		makeAccessibleIfNotNull(this.afterExecute);
-		this.terminated = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "terminated", null);
-		makeAccessibleIfNotNull(this.terminated);
-		this.newTaskForRunnable = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "newTaskFor",
+		this.decorateTaskCallable = makeAccessibleIfNotNullAndOverridden(decorateTaskCallable);
+		Method finalize = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "finalize", null);
+		this.finalize = makeAccessibleIfNotNullAndOverridden(finalize);
+		Method beforeExecute = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "beforeExecute", null);
+		this.beforeExecute = makeAccessibleIfNotNullAndOverridden(beforeExecute);
+		Method afterExecute = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "afterExecute", null);
+		this.afterExecute = makeAccessibleIfNotNullAndOverridden(afterExecute);
+		Method terminated = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "terminated", null);
+		this.terminated = makeAccessibleIfNotNullAndOverridden(terminated);
+		Method newTaskForRunnable = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "newTaskFor",
 				Runnable.class, Object.class);
-		makeAccessibleIfNotNull(this.newTaskForRunnable);
-		this.newTaskForCallable = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "newTaskFor",
+		this.newTaskForRunnable = makeAccessibleIfNotNullAndOverridden(newTaskForRunnable);
+		Method newTaskForCallable = ReflectionUtils.findMethod(ScheduledThreadPoolExecutor.class, "newTaskFor",
 				Callable.class);
-		makeAccessibleIfNotNull(this.newTaskForCallable);
+		this.newTaskForCallable = makeAccessibleIfNotNullAndOverridden(newTaskForCallable);
 	}
 
-	private void makeAccessibleIfNotNull(Method method) {
+	private Method makeAccessibleIfNotNullAndOverridden(Method method) {
 		if (method != null) {
-			ReflectionUtils.makeAccessible(method);
+			if (isMethodOverridden(method)) {
+				try {
+					ReflectionUtils.makeAccessible(method);
+					return method;
+				}
+				catch (Throwable ex) {
+					if (ex.getClass().toString().contains("InaccessibleObjectException")) {
```

Comment:
Wouldn't it be safer to walk down the `getCause() ` calls until we find this or reach the end?